### PR TITLE
fixing monetizationAPIs - testApp

### PR DESCRIPTION
**Pull request by lakhveerk**
Monetization API takes an optional argument planInfo and return error or void. Current behavior assumes it returns planInfo. Fixing this.

### CODE REVIEW

**Comment by erinha** [comment]
$(bundleAnalysisTask.bundleAnalysisComment)

**Comment by erinha** [comment]
$(bundleAnalysisTask.bundleAnalysisComment)

File: apps/teams-test-app/src/components/privateApis/MonetizationAPIs.tsx [hunk 1]
```diff
@@ -12,20 +12,25 @@ const CheckMonetizationCapability = (): React.ReactElement =>
...
+    onClick: {
+      validateInput: planInfo => {
+        if (!planInfo) {
+          return; //This API allow for the input not to be provided
         }
-      };
-
-      monetization.openPurchaseExperience(callback);
-      return 'monetization.openPurchaseExperience()' + noHostSdkMsg;
+        if (typeof planInfo.planId !== 'string') {
```
Comment (MarcinJuraszek) [inline apps/teams-test-app/src/components/privateApis/MonetizationAPIs.tsx:23]: When I was transitioning all the APIs to this new pattern, I decided against checking the type for each of these properties. So I would probably just do:

```
if (!planInfo.planId || !planInfo.term) {
    throw new Error('planId and term are required on input, if provided');
}
```

**Comment by erinha** [comment]
$(bundleAnalysisTask.bundleAnalysisComment)

**Review by MarcinJuraszek — APPROVED** [review]

**Comment by erinha** [comment]
$(bundleAnalysisTask.bundleAnalysisComment)

**Pull request merged**